addons/mail/models/mail_message.py

```diff
@@ -1171,11 +1171,19 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
-        updates = [[
-            (self._cr.dbname, 'res.partner', author.id),
-            {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
-        ] for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id'))]
-        self.env['bus.bus'].sendmany(updates)
+        data_dict = defaultdict(lambda: self.env['mail.message'])
+        for message in messages.sorted('author_id'):
+            partner = self.env.user.partner_id
+            author = message.author_id
+            data_dict[author.id] += message
+            if partner.id not in data_dict:
+                data_dict[partner.id] += message
+        for partner, multimessage in data_dict.items():
+            for message in multimessage:
+                self.env['bus.bus'].sendmany([[
+                    (self._cr.dbname, 'res.partner', partner),
+                    {'type': 'message_notification_update', 'elements': message._message_notification_format()}
+                ]])
```

Comment:
The goal with this part of the code is to call `sendmany` only once. Each partner should lead to 1 bus notification in the array of `sendmany`, and all `messages` of each partner should be in `elements` (notice how it was an array before).